aepp/flowservice.py

```diff
@@ -691,53 +876,103 @@ def createSourceConnectionStreaming(
             name : REQUIRED : Name of the Connection.
             format : REQUIRED : format of the data sent (default : delimited)
             description : REQUIRED : Description of of the Connection Source.
+            spec_name : OPTIONAL : The name of the source specification corresponding to Streaming.
         """
         if self.loggingEnabled:
             self.logger.debug(f"Starting createSourceConnectionStreaming")
+        spec_id = self.getConnectionSpecIdFromName(spec_name)
```

Comment:
I am not sure why you do not hardcode the Sream Spec ID. Could there be more than one Spec ID for Streaming ?

Reply:
Yes good question, I thought about it when creating all the datalake to dlz and. vice versa and found it quite error-prone for myself when I had to hardcode the IDs and I kept mixing them up. I found using the name to be easier to see if I referred to something tied to DLZ or datalake or something else when the IDs just obfuscate it entirely. AFAIK the names are unique I tried creating duplicate ones with same name and it got rejected so there should be no conflict.

Reply:
Hi @cmenguy, I think it is a great user experience to look up by name instead of remembering random ids. But we could have duplicate flow specs names if they are not public flow specs. For a typical usage of destination SDK, the user could create multiple flow specs/connection specs with the same name for testing purposes and only publish one for everyone else to use.

Reply:
I would believe that the user of the library could always default to the `getConnectionSpecs` method in case there are more than one flow with the same name.
I saw that @cmenguy already took care of the situation when calling the `getConnectionSpecIdFromName` method.

```python
raise Exception(f"Expected a single flow specification mapping to flow name '{flow_spec_name}', "
                            f"source spec name '{source_spec_name}' and target spec name '{target_spec_name}'"
                            f"but got {len(flows)}")
```

IMO we are covered. 
We may ponder the requirement to return a list of ID instead of a single string in the future if we see a valid use-case for it but IMO, if there are different ID for a single name, you may want to retrieve the details of the spec to make a choice. 
It would be dangerous to assume any order for the IDs.